docs/dpdk/README.md

```diff
@@ -0,0 +1,74 @@
+# Running DPDK applications in Kubernetes
+
+## Pre-requisites
+
+### Hugepages
+DPDK applications require Hugepages memory. Please refer to the [Hugepages section](http://doc.dpdk.org/guides/linux_gsg/sys_reqs.html#use-of-hugepages-in-the-linux-environment) in DPDK getting started guide on hugespages in DPDK.
+
+Kubernetes nodes can only advertise a single size pre-allocated hugepages. Which means even though one can have both 2M and 1G hugepages in a system, Kubernetes will only recognize the default hugepages as schedulable resources. Workload can request for hugepages using resource requests and limits specifying `hugepage-2Mi` or `hugepage-1Gi` resource references.
```

Comment:
> Is this a limitation of kubernetes when handling hugepages?

It is indeed unfortunately. Please see note below taken from [here](https://kubernetes.io/docs/tasks/manage-hugepages/scheduling-hugepages/#before-you-begin)

 _1. Kubernetes nodes must pre-allocate huge pages in order for the node to report its huge page capacity. A node may only pre-allocate huge pages for a **single size**._


> I'm only aware of that one pod can not use two sizes of hugepages,
> but this seems to say that we have cannot have two pods on the same node, using different hugepage size?

 If I understand it correctly from previous statement, since a node can only pre-allocate a single size hugepage, it is not possible to schedule pods with different hugepages on same node.